tests/test_operator.py

```diff
@@ -467,26 +467,26 @@ def test_override_timefunction_data(self):
 
         # Run with default value
         a.data[:] = 1.
-        op(t=2)
-        assert (a.data[:] == 4.).all()
+        op(t=1)
+        assert (a.data[0] == 4.).all()
 
         # Override with symbol (different name)
         a1 = TimeFunction(name='a1', grid=grid)
         a1.data[:] = 2.
-        op(t=2, a=a1)
-        assert (a1.data[:] == 5.).all()
+        op(t=1, a=a1)
+        assert (a1.data[0] == 5.).all()
 
         # Override with symbol (same name as original)
         a2 = TimeFunction(name='a', grid=grid)
         a2.data[:] = 3.
-        op(t=2, a=a2)
-        assert (a2.data[:] == 6.).all()
+        op(t=1, a=a2)
+        assert (a2.data[0] == 6.).all()
 
         # Override with user-allocated numpy data
         a3 = np.zeros_like(a.data)
         a3[:] = 4.
-        op(t=2, a=a3)
-        assert (a3[:] == 7.).all()
+        op(t=1, a=a3)
+        assert (a3[0] == 7.).all()
```

Comment:
Why did these ones change (`t`, indices `mask`)?

Reply:
perhaps unnecessary, I might have changed it only for homogeneity with the previous lines. I'll check that

Reply:
OK so if you leave ``t=2``, then you get:
- ``a3[0] == 10``
- ``a3[1] == 4``
this is because of what I explained in the PR description. IOW, there is no stencil in time for this equation, so the buffer slot 0 is the only one used (ie, updates occur in ``a3[0]`` only). If you do ``t=2``, you bring it from 4 to 10; with ``t=1`` (as here) we go from 4 to 7. ``a3[1]`` stays at 4. I'll assert that

Reply:
The stencil is `u(t) = u(t) + 3`  with everything initialized at 4.
`t=1` => `u[0] =7` the rest still at 4
`t=2` =>  `u[0] = u[1] = 7` the rest still at 4

and so on. If you get
`a3[0] == 10`
something is not right.

Reply:
there are only two buffer slots, and ``u[1]`` is never incremented. So I really don't understand what's the issue here. My assert checks against 7 after one timestep, so...

Reply:
The generated code implements `u(t) = u(t-1) + 3`, not `u(t) = u(t) +3`. 

If `u` is initialized with `3` everywhere, every time, then no matter how many time steps you run (including `t=2`, then `u` can only be equal to 7.

Reply:
Actually, thinking about it, this stencil cannot be implemented without solving the full time history.

Reply:
> The generated code implements u(t) = u(t-1) + 3, not u(t) = u(t) +3

??

isn't the Operator:
```
op = Operator(Eq(a, a + 3), dle=None)
```

Reply:
it is but the C code is `u[t1] = u[t1] + 3` (ot t0 don;t rememeber with `t1=time%1`.

So at the second time step, `u[t1]` contains u(t=0) and does `u[t1 (t=1)] = u[t1 (t=0)] + 3`.

Reply:
> The generated code implements u(t) = u(t-1) + 3, not u(t) = u(t) +3.

I checked again. Here's the generated code:
```
int Kernel(float *restrict a_vec, const int time_e, const int time_s, void *_timers, const int x_e, const int x_s, const int x_size, const int y_e, const int y_s, const int y_size, const int z_e, const int z_s, const int z_size)
{
  float (*restrict a)[x_size][y_size][z_size] __attribute__((aligned(64))) = (float (*)[x_size][y_size][z_size]) a_vec;
  struct profile *timers = (struct profile*) _timers;
  for (int time = time_s, t0 = (time)%(1); time < time_e; time += 1, t0 = (time)%(1))
  {
    struct timeval start_section_0, end_section_0;
    gettimeofday(&start_section_0, NULL);
    for (int x = x_s; x < x_e; x += 1)
    {
      for (int y = y_s; y < y_e; y += 1)
      {
        for (int z = z_s; z < z_e; z += 1)
        {
          a[t0][x][y][z] = a[t0][x][y][z] + 3;
        }
      }
    }
    gettimeofday(&end_section_0, NULL);
    timers->section_0 += (double)(end_section_0.tv_sec-start_section_0.tv_sec)+(double)(end_section_0.tv_usec-start_section_0.tv_usec)/1000000;
  }
  return 0;
}
```

I'm not sure why you keep saying the generated code does not implement ``u[t] = u[t] + 3``. It does. ``t_s`` starts at 0 and ramps up to 1, given that I imposed ``op.apply(t=1, ...``). So ``a3[0]`` increases by 3, while ``a3[1]`` stays at 4.